components/admin_console/team_channel_settings/channel/details/channel_details.tsx

```diff
@@ -440,19 +440,19 @@ export default class ChannelDetails extends React.PureComponent<ChannelDetailsPr
             filter((g) => {
                 return origGroups.some((group) => group.id === g.id && group.scheme_admin !== g.scheme_admin);
             }).
-            map((g) => actions.patchGroupSyncable(g.id, channelID, Groups.SYNCABLE_TYPE_CHANNEL, {scheme_admin: g.scheme_admin}));
+            map((g) => actions.patchGroupSyncable(g.id, channelID, 'channel', {scheme_admin: g.scheme_admin, auto_add: false}));
```

Comment:
If I use `Groups.SYNCABLE_TYPE_CHANNEL` constant I have following error : `error TS2345: Argument of type 'string' is not assignable to parameter of type 'SyncableType'` . Seems like `SyncableType` is a String Literal Type which only accepts 'channel' or 'team' values. From what I understand constant declaration of `SYNCABLE_TYPE_CHANNEL` should be changer to something like : 
```
declare const _default: {
    SYNCABLE_TYPE_TEAM: 'team';
    SYNCABLE_TYPE_CHANNEL: 'channel';
};
export default _default;
```
Problem is these constants are generated in webapp project, I think it should imply a change in mattermost-redux project. What do you think?

Reply:
Unfortunately I just tested your solution and it produces the same error.
Are you able to do a simple cast in this case?
```suggestion
            map((g) => actions.patchGroupSyncable(g.id, channelID, Groups.SYNCABLE_TYPE_CHANNEL as SyncableType, {scheme_admin: g.scheme_admin, auto_add: false}));
```

Reply:
Strange, it was working on my side by fixing directly the file in `node_module`/`mattermost-redux`/`constants/groups.d.ts`. IMHO it would be better to fix the constant directly to avoid such cast in future, but I'm not expert in this area so as you want!

Reply:
Looks like I misread the error...sorry about that. This seems like something we should do across the board, I'm going to look into it today. Thanks for your input :)

Reply:
https://github.com/mattermost/mattermost-redux/pull/1284

Reply:
@cinlloc Above PR is merged.

Reply:
@devinbinnie thx, it fixes this issue. Unfortunately, I have another issue now which would need following PR: https://github.com/mattermost/mattermost-redux/pull/1291. To be continued...

Reply:
@cinlloc Above is merged. We're getting there.

Reply:
@cinlloc Can we get this fixed?

Reply:
yep, i'm on it

Reply:
@devinbinnie fixed in 46ae22d477